CONTRIBUTIONS.md

```diff
@@ -97,6 +97,9 @@ A lot of people are excited about building a JSON API in Servant and everyone se
   - Investigate the tradeoff of doing all the implementation in haskell vs. interface, via haskell, to html templates.
   - Investigate how to integrate with existing jQuery widgets (calendar, accordion, search/sort tables, editable grids, etc)
   - Forms relying only on server side validation or with mixed client/server side validation
+  - Investigate the preferred way to architecture an application. How powerful, and how general, is it?
```

Comment:
Yes, I plan to describe all the work I did in this area (the App datatype, with one constructor for each page of the app). That also might be turned into a tiny library, but I have to get an efficiency imprimatur from Ryan first.